farmer|harvester|server: Fix harvester cache updates, bump message size, increase timeout

xdustinface opened the pull request:
See commit messages.

xdustinface committed:
farmer: Increase timeout for `HarvesterAPI.request_plots` requests

Set it to `UPDATE_HARVESTER_CACHE_INTERVAL`.

xdustinface committed:
farmer|harvester: Add some debug logs around `get_harvester` stuff

tnt-codie commented:
If the plots are requested once per minute why do you choose 20 seconds as timeout? Is there a reason why we won't choose simply the same minus some tolerance such as 55s?

xdustinface committed:
server: Bump size limit for `respond_plots` from 10MB/msg to 100MB/msg

Those messages can become pretty big now for large farms.

xdustinface committed:
farmer: Keep also track of failed cache update attempts

Prior to this commit the farmer called the `request_plots` every second 
for each harvester as long as they failed to respond properly. Since the 
rate limit was 10/minute this lead to hitting the rate limit if the 
harvester didn't responds for 10 tries in a row for whatever reason.

This commit changes the behaviour to always keep track of request 
attempts even if they end up in a timeout to really only re-try every 
60s no matter what.

xdustinface committed:
farmer: Introduce and use `HarvesterCacheEntry`

xdustinface commented:
> If the plots are requested once per minute why do you choose 20 seconds as timeout? Is there a reason why we won't choose simply the same minus some tolerance such as 55s?

Right, i actually don't see a reason to not increase it even more. I set it to `UPDATE_HARVESTER_CACHE_INTERVAL` see 76d26f8.

emlowe reviewed (approved):
Looks good and is testing well by affected users
